client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala

```diff
@@ -300,6 +301,14 @@ class LifecycleManager(val appUniqueId: String, val conf: CelebornConf) extends
       logDebug(s"Received GetShuffleFileGroup request," +
         s"shuffleId $shuffleId.")
       handleGetReducerFileGroup(context, shuffleId)
+
+    case pb: PbHeartbeatFromClient =>
+      val shuffleIds = new util.ArrayList[Integer](pb.getShuffleIdList)
+      val pred = new Predicate[Integer] {
+        override def test(id: Integer): Boolean = registeredShuffle.contains(id)
+      }
+      shuffleIds.removeIf(pred)
+      context.reply(HeartbeatFromClientResponse(shuffleIds))
```

Comment:
Maybe better add a method handleHeartbeatFromClient?

Reply:
Right, although it's pretty simple right now, in the future it can be more complicated.